src/KoIdentity.Abstraction/Controllers/IUserController.cs

```diff
@@ -0,0 +1,175 @@
+// Copyright © Tekoding 2021. All rights reserved.
+// Author: TheRealLenon
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Tekoding.KoIdentity.Abstraction.Controllers
+{
+    /// <summary>
+    /// Provides an abstraction for API´s used by an user controller.
+    /// </summary>
+    /// <typeparam name="TUserDto">The type encapsulating a data transfer object of a user.</typeparam>
+    /// <typeparam name="TRole">The type encapsulating a data transfer object of a role.</typeparam>
+    /// <typeparam name="TRegistrationDto">
+    /// The Type encapsulating a data transfer object being used for registering a user.
+    /// </typeparam>
+    /// <typeparam name="TLoginDto">
+    /// The type encapsulating a data transfer object being used for signing in a user.
+    /// </typeparam>
+    public interface IUserController<TUserDto, TRole, in TRegistrationDto, in TLoginDto>
+        where TRegistrationDto : class
+        where TLoginDto : class
+    {
+        /// <summary>
+        /// Registers a user onto the platform.
+        /// </summary>
+        /// <param name="registrationDto">The data transfer being used for registering a user.</param>
+        /// <param name="cancellationToken">
+        /// The <see cref="CancellationToken"/> is being used to propagate notifications that the operation should be
+        /// canceled.
+        /// </param>
+        /// <returns>
+        /// The <see cref="Task{TResult}"/> that represents the asynchronous operation, containing the
+        /// <see cref="string"/> that contains information if the registration succeeded.
+        /// </returns>
+        public Task<string> Register(TRegistrationDto registrationDto, CancellationToken cancellationToken = default);
```

Comment:
Please change to the following
```suggestion
        public Task<string> RegisterUser(TRegistrationDto registrationDto, CancellationToken cancellationToken = default);
```